src/compiler/codegen.js

```diff
@@ -117,21 +119,10 @@ function genData (el: ASTElement): string {
   if (el.slotTarget) {
     data += `slot:${el.slotTarget},`
   }
-  // class
-  if (el.staticClass) {
-    data += `staticClass:${el.staticClass},`
-  }
-  if (el.classBinding) {
-    data += `class:${el.classBinding},`
-  }
-  // style
-  if (el.styleBinding) {
-    data += `style:${el.styleBinding},`
-  }
-  // transition
-  if (el.transition) {
-    data += `transition:{definition:(${el.transition}),appear:${el.transitionOnAppear}},`
-  }
+  // platform modules
+  platformModules.forEach(module => {
+    data = module.genData(el, data)
```

Comment:
I think it's better to do `data += module.genData(el)` here. `genData` for each module doesn't need existing `data` as an argument.